### PR TITLE
Added `var<mat>` implementation of `log_determinant_spd` (Issue #2101)

### PR DESCRIPTION
**Pull request by bbbales2**
## Summary

Added `var<mat>` implementation of `log_determinant_spd`

## Release notes

Added `var<mat>` implementation of `log_determinant_spd`

## Checklist

- [x] Math issue #2101

- [x] Copyright holder: Columbia University

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [x] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

**Comment by bbbales2** [comment]
Ready for review

File: stan/math/rev/fun/log_determinant_spd.hpp [hunk 1]
```diff
@@ -63,6 +63,48 @@ inline var log_determinant_spd(const EigMat& m) {
...
+  arena_t<Eigen::MatrixXd> arena_m_inv_transpose
+      = Eigen::MatrixXd::Identity(m.rows(), m.cols());
+  ldlt.solveInPlace(arena_m_inv_transpose);
+
+  var log_det = sum(log(ldlt.vectorD()));
+
+  reverse_pass_callback([m, log_det, arena_m_inv_transpose]() mutable {
+    m.adj() += log_det.adj() * arena_m_inv_transpose;
+  });
+  return log_det;
```
Comment (SteveBronder) [inline stan/math/rev/fun/log_determinant_spd.hpp:105]: You could use `make_callback_var/vari` here but I'm also fine with `reverse_pass_callback()`

**Review by SteveBronder — APPROVED** [review]
Looks good! One optional suggestion but besides that lgtm!

**Comment by bbbales2** [comment]
@SteveBronder this is ready when tests passes. Took me two tries to get the `make_callback_var` right :/.

**Comment by bbbales2** [comment]
@SteveBronder this should be good to go. Only changes I did since your last review was swapping to `make_callback_var`

**Review by SteveBronder — APPROVED** [review]
Looks good!

**Pull request merged**